### PR TITLE
fix: set root kwargs for parity trace at end of builder-logic

### PR DESCRIPTION
**Pull request by unparalleled-js**
### What I did

To more honorably honor the overrides and actually have them work, they should be set _after_ we build up the object from the call data.

Additionally, fixed many small things in the code base.

### How I did it

Adds in the `root_kwargs` at end of function instead of beginning.

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

**Review by fubuloubu — APPROVED** [review]

File: evm_trace/base.py [hunk 1]
```diff
@@ -19,6 +19,10 @@ class TraceFrame(BaseModel):
     memory: List[HexBytes] = []
     storage: Dict[HexBytes, HexBytes] = {}
 
+    @validator("pc", "gas", "gas_cost", "depth", pre=True)
+    def validate_ints(cls, value):
```
Comment (helloibis) [inline evm_trace/base.py:23]: Does Pydantic not have an issue if someone creates a `TraceFrame` with a str for fields like `pc` or `gas`?

**Review by helloibis — APPROVED** [review]
:shipit:

File: evm_trace/base.py (hunk 1 above)
Comment (unparalleled-js) [reply evm_trace/base.py:23]: It does! 
Unless you have it get handled in validator, like it is here. 

This happens on Nethermind clients, according to some comment in Bronwnie, so it doesn't hurt to have this validator

File: evm_trace/parity.py [hunk 2]
```diff
@@ -153,17 +156,19 @@ def get_calltree_from_parity_trace(
             )
 
     elif root.call_type == CallType.SELFDESTRUCT:
-        selfdestruct_action: SelfDestructAction = root.action  # type: ignore
+        selfdestruct_action: SelfDestructAction = cast(SelfDestructAction, root.action)
```
[the remark this answers is not in the document]
Comment (unparalleled-js) [reply evm_trace/parity.py:159]: Dang overloaded `self`!

**Pull request merged**